File: server/src/entity/Task.ts
import { Priority } from "../model/Priority";
import { BaseEntity, Column, CreateDateColumn, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn, UpdateDateColumn } from "typeorm";
import { User } from "./User";


@Entity("task")
export class Task extends BaseEntity {

    @PrimaryGeneratedColumn()
    id: number

    @Column()
    title: string

    @Column({name: "is_completed", default: false})
    isCompleted: boolean

    @Column()
    description: string

    @Column({
        type: "enum",
        enum: Priority
    })
    priority: Priority

    @ManyToOne(() => User, user => user.tasks)
    @JoinColumn({ name: "creator_id" })
    creator: User

    @Column({name: "creator_id"})
    creatorId: string

    @CreateDateColumn({name: "created_at"})
    createdAt: Date

    @UpdateDateColumn({ name: "updated_at"})
    updatedAt: Date
}

File: client/app/src/main/java/com/ben/taskplanner/model/task/Task.kt
package com.ben.taskplanner.model.task


import android.os.Parcelable
import com.ben.taskplanner.model.Priority
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Task(
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("creator_id")
    val creatorId: String,
    val description: String,
    val id: Int,
    @SerializedName("is_completed")
    val isCompleted: Boolean,
    val name: String,
    val priority: String,
    val surname: String,
    val title: String,
    @SerializedName("updated_at")
    val updatedAt: String
): Parcelable